cmd/osbuild-composer/composer.go

```diff
@@ -0,0 +1,268 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"path"
+
+	"github.com/osbuild/osbuild-composer/internal/common"
+	"github.com/osbuild/osbuild-composer/internal/jobqueue/fsjobqueue"
+	"github.com/osbuild/osbuild-composer/internal/kojiapi"
+	"github.com/osbuild/osbuild-composer/internal/rpmmd"
+	"github.com/osbuild/osbuild-composer/internal/store"
+	"github.com/osbuild/osbuild-composer/internal/upload/koji"
+	"github.com/osbuild/osbuild-composer/internal/weldr"
+	"github.com/osbuild/osbuild-composer/internal/worker"
+
+	"github.com/osbuild/osbuild-composer/internal/distro"
+	"github.com/osbuild/osbuild-composer/internal/distro/fedora31"
+	"github.com/osbuild/osbuild-composer/internal/distro/fedora32"
+	"github.com/osbuild/osbuild-composer/internal/distro/rhel8"
+)
+
+type Composer struct {
+	config   *ComposerConfigFile
+	stateDir string
+	cacheDir string
+	logger   *log.Logger
+	distros  *distro.Registry
+
+	rpm rpmmd.RPMMD
+
+	workers *worker.Server
+	weldr   *weldr.API
+	koji    *kojiapi.Server
+
+	weldrListener, localWorkerListener, workerListener, kojiListener net.Listener
+}
+
+func NewComposer(config *ComposerConfigFile, stateDir, cacheDir string, logger *log.Logger) (*Composer, error) {
```

Comment:
The canonical name would just be `New()`, wouldn't it?

Reply:
It would, but this is not a package. Having a blank `New(...)` in `main.go` seemed confusing to me.